sg/fn.go

```diff
@@ -70,8 +74,14 @@ func newFn(f interface{}, args ...interface{}) (Function, error) {
 	if hasNamespace {
 		argCount++ // +1 for the namespace
 	}
+	argsID, err := json.Marshal(args)
+	if err != nil {
+		return nil, fmt.Errorf("failed to generate JSON name for args: %w", err)
+	}
+	name := runtime.FuncForPC(reflect.ValueOf(f).Pointer()).Name()
 	return fn{
-		name: runtime.FuncForPC(reflect.ValueOf(f).Pointer()).Name(),
+		name: name,
+		id:   name + "(" + string(argsID) + ")",
```

Comment:
Is this the name of the arguments or the actual values passed? Caused the former won't solve the problem right?

Reply:
actual values passed, so this should do the trick